kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/xstream/XStreamMarshaller.java

```diff
@@ -152,18 +148,7 @@ protected void configureMarshaller(Set<Class<?>> classes,
         String[] voidDeny = {"void.class", "Void.class"};
         this.xstream.denyTypes(voidDeny);
 
-        String packageList = System.getProperty(SYSTEM_XSTREAM_ENABLED_PACKAGES,
-                                                DEFAULT_PACKAGES_WILDCARD);
-
-        String[] filter;
-
-        if (StringUtils.isEmpty(packageList)) {
-            filter = new String[]{DEFAULT_PACKAGES_WILDCARD};
-        } else {
-            filter = packageList.split(",");
-        }
-
-        this.xstream.addPermission(new WildcardTypePermission(filter));
+        this.xstream.addPermission(new KieServerTypePermission(classes));
```

Comment:
@mswiderski is it possible that the Set classes is null in some circumstances? KieServerTypePermission is now failing with a NPE at line 56 and this seems that only case where the Set is passed from the outside.